src/js/ui/night/controls/brush-control.js

```diff
@@ -0,0 +1,445 @@
+   * This will also stop the painting phase
+   * @param  {[type]} e [description]
+   * @return {[type]}   [description]
+   */
+  _onMouseLeave (e) {
+    this._painting = false
+    this._hideCursor()
+  }
+
+  /**
+   * Adds a control point to the path.
+   * Also the status of the mouse button, i.e. pressed or not, will be logged
+   * @param {Vector2} position
+   * @param {Boolean} mouseButtonPressed = false
+   */
+  _addControlPoint (position, mouseButtonPressed = false) {
+    var controlPoints = this._operation.getControlPoints()
+    controlPoints.push(position)
+    this._operation.setControlPoints(controlPoints)
+
+    var buttonStatus = this._operation.getButtonStatus()
+    buttonStatus.push(mouseButtonPressed)
+    this._operation.setButtonStatus(buttonStatus)
+  }
+
+  /**
+   * Adds the current selected color to the array of colors of the operation.
+   * That color will be used to paint the next stroke
+   */
+  _addCurrentColor () {
+    var colors = this._operation.getColors()
+    colors.push(this._currentColor.clone())
+    this._operation.setColors(colors)
+  }
+
+  /**
+   * Adds the current selected thickness to the array of thicknesses of the operation.
+   * That thickness will be used to paint the next stroke
+   */
+  _addCurrentThickness () {
+    var thicknesses = this._operation.getThicknesses()
+    thicknesses.push(this._currentThickness)
+    this._operation.setThicknesses(thicknesses)
+  }
+
+  /**
+   * Triggers a path-redraw
+   */
+  _redrawPath () {
+    this._ui.canvas.render()
+  }
+
+  /**
+   * Calculates the mouse position, relative to the upper-left corner
+   * of the canvas
+   * @param  {Event} e
+   * @return {Vector2} The Mouse Position
+   */
+  _getRelativeMousePositionFromEvent (e) {
+    var clientRect = this._container.getBoundingClientRect()
+    var offset = new Vector2(clientRect.left, clientRect.top + document.body.scrollTop)
+    var absolutePosition = Utils.getEventPosition(e).subtract(offset)
+    return absolutePosition.divide(this._ui.canvas.size)
+  }
+
+  /**
+   * Gets called when the thickness has been changed
+   * @override
+   */
+  _onThicknessUpdate (value) {
+    this._currentThickness = value / this._getLongerSideSize()
+    this._highlightDoneButton()
+    this._setCursorSize(this._currentThickness * this._getLongerSideSize())
+  }
+
+  /**
+   * Gets called when the color has been changed
+   * @override
+   */
+  _onColorUpdate (value) {
+    this._currentColor = value
+    this._highlightDoneButton()
+    this._setCursorColor(value)
+  }
+
+  /**
+   * Returns the longer size of the ui canvas
+   * @return {Number}
+   */
+  _getLongerSideSize () {
+    return this._ui.canvas.size.x > this._ui.canvas.size.y ? this._ui.canvas.size.x : this._ui.canvas.size.y
+  }
+
+  /**
+   * Moves our custom cursor to the specified position
+   * @param  {Vector2} position
+   */
+  _moveCursorTo (position) {
+    let myCursor = this._canvasControls.querySelector('#mycursor')
```

Comment:
`querySelector` is slow - since this is a method that is called a lot while drawing, please make sure you initially get the cursor in `_onEnter` and use it in all other methods